rmw_connext_dynamic_cpp/src/functions.cpp

```diff
@@ -286,33 +340,42 @@ rmw_create_publisher(
   status = participant->get_default_publisher_qos(publisher_qos);
   if (status != DDS_RETCODE_OK) {
     rmw_set_error_string("failed to get default publisher qos");
-    // printf("get_default_publisher_qos() failed. Status = %d\n", status);
     return NULL;
   }
 
   DDSPublisher * dds_publisher = participant->create_publisher(
     publisher_qos, NULL, DDS_STATUS_MASK_NONE);
   if (!dds_publisher) {
-    rmw_set_error_string("  create_publisher() could not create publisher");
+    rmw_set_error_string("failed to create publisher");
     return NULL;
   }
 
 
-  DDS_TopicQos default_topic_qos;
-  status = participant->get_default_topic_qos(default_topic_qos);
-  if (status != DDS_RETCODE_OK) {
-    rmw_set_error_string("failed to get default topic qos");
-    // printf("get_default_topic_qos() failed. Status = %d\n", status);
-    return NULL;
-  }
+  DDSTopic * topic;
+  DDSTopicDescription * topic_description = participant->lookup_topicdescription(topic_name);
+  if (!topic_description) {
+    DDS_TopicQos default_topic_qos;
+    status = participant->get_default_topic_qos(default_topic_qos);
+    if (status != DDS_RETCODE_OK) {
+      rmw_set_error_string("failed to get default topic qos");
+      return NULL;
+    }
 
-  DDSTopic * topic = participant->create_topic(
-    topic_name, type_name.c_str(), default_topic_qos, NULL,
-    DDS_STATUS_MASK_NONE
-    );
-  if (!topic) {
-    rmw_set_error_string("  create_topic() could not create topic");
-    return NULL;
+    topic = participant->create_topic(
+      topic_name, type_name.c_str(), default_topic_qos, NULL,
+      DDS_STATUS_MASK_NONE
+      );
```

Comment:
[style] As far as I understand our style guide, this should either have two less spaces in front of it or it should be on the previous line.